### PR TITLE
Setting up Github codespaces

### PR DESCRIPTION
**Pull request by cesy**
The default doesn't include any database, so we need a devcontainer to include our standard stuff. This one from Microsoft has a Docker file that has Postgres set up for us.

### CODE REVIEW

**Comment by Br3nda** [comment]
This looks useful. Are there docs on where to get started trying this out?

**Comment by CloCkWeRX** [comment]
https://github.com/cesy/growstuff/tree/codespaces -> 

image

**Comment by CloCkWeRX** [comment]
Minor tweaks needed - default postgres connection? (db:create was angry, db:migrate worked)
```
Caused by:
PG::ConnectionBad: could not connect to server: No such file or directory
        Is the server running locally and accepting
        connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
/usr/local/rvm/gems/default/gems/pg-1.4.5/lib/pg/connection.rb:755:in `connect_start'
```
And ruby 3.1.2 vs 3.1.3

And elasticsearch?
== 20191226051019 ElasticIndexing: migrating ==================================
-- indexing crops
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Failed to open TCP connection to localhost:9200 (Cannot assign requested address - connect(2) for "localhost" port 9200)
/workspaces/growstuff/db/migrate/20191226051019_elastic_indexing.rb:6:in `up'

Caused by:
Faraday::ConnectionFailed: Failed to open TCP connection to localhost:9200 (Cannot assign requested address - connect(2) for "localhost" port 9200)
/workspaces/growstuff/db/migrate/20191226051019_elastic_indexing.rb:6:in `up'

Caused by:
Errno::EADDRNOTAVAIL: Failed to open TCP connection to localhost:9200 (Cannot assign requested address - connect(2) for "localhost" port 9200)
/workspaces/growstuff/db/migrate/20191226051019_elastic_indexing.rb:6:in `up'

Caused by:
Errno::EADDRNOTAVAIL: Cannot assign requested address - connect(2) for "localhost" port 9200
/workspaces/growstuff/db/migrate/20191226051019_elastic_indexing.rb:6:in `up'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+// Features to add to the dev container. More info: https://containers.dev/features.
+// Need something to get javascript working
+"features": {
+    "ghcr.io/devcontainers/features/node:1": {}
+},
+
+// Use 'forwardPorts' to make a list of ports inside the container available locally.
+// This can be used to network with other containers or the host.
+"forwardPorts": [3000, 5432]
```
Comment (CloCkWeRX) [inline .devcontainer/devcontainer.json]:
```suggestion
"forwardPorts": [3000, 5432, 9200]
```

**Comment by CloCkWeRX** [comment]
Annoying, I don't know docker quite well enough to say "app depends on es01 being up" and "es01 should be visible to app".

Also
Supposedly, we should be able to pass in `VARIANT` to explicitly say 'ruby 3.1.2'
https://github.com/microsoft/vscode-dev-containers/blob/main/containers/ruby/.devcontainer/base.Dockerfile#L3

**Comment by CloCkWeRX** [comment]
Have to fight permissions here, show creating a different branch in the main org.

**Pull request closed without merging**